### DISCUSSION TITLE
Disable binary detection

### DISCUSSION THREAD

**Discussion by yosiat** [socketio/socket.io#3826] [Q&A] [answered]
Hi!

I am upgrading to latest version of socket.io and saw that `binary` method is removed on socket.io (https://github.com/socketio/socket.io/commit/669592d120409a5cf00f128070dee6d22259ba4f)

I used to pass to all of my emits, `binary(false)` so save the `hasBinary` check which came hot in my CPU profiles.

Is there any way to disable it while using socket.io-parser ? or I need to fork it?

**Comment by darrachequesne** [comment, marked as answer]
> Is there any way to disable it while using socket.io-parser

No, there is currently no way to disable it. You can provide your own parser though, please see the documentation here: https://socket.io/docs/v3/custom-parser/#Implementing-your-own-parser
